script/hassfest/quality_scale_rules/config_flow.py

```diff
@@ -0,0 +1,43 @@
+"""Enforce config_file quality scale rules.
+
+This validates that the integration has a config flow, which is required
+to set up the integration from the UI.
+"""
+
+import json
+
+from homeassistant.generated import config_flows
+
+from . import QualityScaleCheck
+
+
+def validate(check: QualityScaleCheck) -> None:
+    """Validate that the integration has a config flow."""
+    flows = config_flows.FLOWS
+    domains = flows["integration"]
+    if check.integration.domain not in domains:
+        check.add_error(
+            "config_flow",
+            "Integration is missing a config_flow.py, and needs to be set up with the UI",
+        )
+
+    # Validate config flow translations
+    strings_file = check.integration.path / "strings.json"
+    try:
+        strings = json.loads(strings_file.read_text())
+    except ValueError:
+        # Assume that other checks will catch invalid JSON
+        return
+
+    # This assumes that all fields will have translations for field names, then
+    # checks for data_description translations.
+    for step, step_translations in strings.get("config", {}).get("step", {}).items():
+        data_description = step_translations.get("data_description", {})
+        for field in step_translations.get("data", {}):
+            if field in data_description:
+                continue
```

Comment:
This assumes that data is already there...
I was thinking to tweak the translation checks in `tests\components\conftest.py` instead

Reply:
Yes, my assumption is that people don't forget to add these since its too obvious as they show up weird in the UI and they are caught in code review.  However data description is easier to miss, such all the examples caught currently.

Adding tests following the existing translation tests sounds good to me, and I was thinking about using that for other rules that can't be enforced with hassfest anyway. I'm fine to move this check to the translations since we already have infra there.

Generally does it make sense to use a combination of both approaches to enforce these rules?  (for this case of translations, happy to move this to tests -- we can assume config flow is 100% covered by tests)